### ISSUE TITLE
[BUG] wasi:logging is not compatible with wit-deps and docs missed info about wit-deps

### ISSUE THREAD

**Issue by deem0n** [wasmCloud/wasmCloud#3287]
### Affected project(s)

- [X] documentation
- [ ] examples
- [ ] wasmCloud host
- [ ] wasmCloud CLI (wash)
- [ ] wasmCloud dashboard UI (washboard)
- [ ] capability providers
- [ ] provider bindgen
- [ ] control interface client
- [X] other / not sure

### Describe the bug

All examples and docs contain wit examples as follows:

wit/world.wit
```
world example {
  import wasi:logging/logging;
}
```

wit/deps.toml
```
logging = "https://github.com/WebAssembly/wasi-logging/archive/main.tar.gz"
```

wit/deps/logging/world.wit
```
package wasi:logging;

world imports {
    import logging;
}
```

That should work with `wash build` (and wokls fine in wasmcloud 1.3, i.e. you will see logs from your component in wasmcloud log output)

### Steps to reproduce

If you want to add `wasi:messaging` to your existing project or will run `wit-deps update` for other reason, you will get broken logs behavior. That is becouse `wit-deps-cli 0.4.0` will include logging package version: 

wit/deps/logging/world.wit
```
package wasi:logging@0.1.0-draft;

world imports {
    import logging;
}
```

To be able to compile you probably will amend wit/world.wit

```
world example {
  import wasi:logging/logging@0.1.0-draft;
}
```

If you compile now with `wash build` you will get:

- wasmcloud 1.2: hangs or no logs from your component.
- wasmcloud 1.3: errors in main log like this:

```
2024-10-10T16:14:33.497571Z  WARN wasmcloud_host::wasmbus: failed to handle invocation err=failed to serve `wrpc:http/incoming-handler@0.1.0.handle` invocation

Caused by:
    0: failed to call `wasi:http/incoming-handler.handle`
    1: error while executing at wasm backtrace:
           0: 0x2c263 - wit-component:shim!indirect-wasi:logging/logging@0.1.0-draft-log
           1: 0x45a8 - imei_injector.wasm!wasi:http/incoming-handler@0.2.0#handle
    2: failed to invoke `wasi:logging/logging@0.1.0-draft.log` polyfill via wRPC
    3: failed to call `log` in instance `wasi:logging/logging@0.1.0-draft` (failed to find a configured link with name `default` from component `imei_injector-injector`, please check your configuration)
```

### Expected behavior

Obviously everything should work without manual editing after `wit-deps update` without component hangs and logs must arrive in the log file.

I have no idea how and where to fix it, but will happy to fix documentation and add info about required step with `wit-deps` as it seems missed in many places. It will not resolve the root problem, though. If docs fix is welcome, just assiign me to this issue.

### Environment

- OS: ANY
- Shell ANY
- wasmcloud 1.2 and 1.3
- wit-deps-cli 0.4.0

**Comment by brooksmtownsend** [comment]
Hey @deem0n thank you for filing this issue, not versioning `wasi:logging/logging` was certainly an oversight to start. We just added support for both the versioned and unversioned interface in https://github.com/wasmCloud/wasmCloud/pull/3281, so once that's merged + cut this will work

**Comment by deem0n** [comment]
That's great !

What adding info about requirement to run `wit-deps update`? At least on this page https://wasmcloud.com/docs/tour/adding-capabilities

Editing `wit/world.wit` without running `wit-deps update` is not enough I guess.

**Comment by atifsyedali** [comment]
There is a similar problem with `wasi:keyvalue`... see https://github.com/WebAssembly/wasi-keyvalue/issues/50

If you pull the keyvalue WIT in your component, you get `wasi:keyvalue@0.2.0-draft2` (note the `2` at the end). But the latest `ghcr.io/wasmcloud/keyvalue-redis` is still on `wasi:keyvalue@0.2.0-draft` and just trying to do any keyvalue operations then fails.

**Comment by deem0n** [comment]
I would like to have `wish providers list` to see all exact versions from the running host then.

**Comment by lachieh** [comment]
I believe `wash providers list` will give you this information, @deem0n. If not, it'd be great to capture that as a separate issue.

**Comment by deem0n** [comment]
I see this for `get inventory`

```
  Host ID                                                            Friendly name
  NCR5TEJFPJSQSSNCKK67B7EOJXTZJO22IAAWD37Z4LVSXXC23OTGNCI4           solitary-hamster-7962

  Host labels
  hostcore.arch                       x86_64
  hostcore.os                         macos
  hostcore.osfamily                   unix

  Component ID                        Name                           Max count
  echo                                echo-messaging                 1
  imei_injector-injector              imei-injector                  1

  Provider ID                         Name
  rust_echo_messaging-nats            messaging-nats-provider
  imei_injector-messaging_nats        messaging-nats-provider
  imei_injector-httpserver_injector   http-server-provider
```

1. There are no built-in providers (i.e. logging)
2. There are no exact versions shown

:-(

**Comment by deem0n** [comment]
Also, I found 3d provider which is affected with this issue: config. All of them missed exact version number in deps.toml:

```
keyvalue = "https://github.com/WebAssembly/wasi-keyvalue/archive/main.tar.gz"
logging = "https://github.com/WebAssembly/wasi-logging/archive/main.tar.gz"
config = "https://github.com/WebAssembly/wasi-runtime-config/archive/main.tar.gz"
```

**Comment by deem0n** [comment]
and blobstore: `blobstore = "https://github.com/WebAssembly/wasi-blobstore/archive/main.tar.gz"`

**Comment by rvolosatovs** [comment]
You should always pin to exact tags/releases/commits when using `wit-deps` - if our examples do not do that - we should fix that

**Comment by deem0n** [comment]
I see 2 problems for newby wasmcloud developer:

- How do I know exact versions for builtin providers, `logging` for example. Should I always dig into github sources?
- If I know exact version numbers, how do I construct links for `wit/deps.toml` ? Is there any rule or hint?

**Comment by rvolosatovs** [comment]
> I see 2 problems for newby wasmcloud developer:
> 
>     * How do I know exact versions for builtin providers, `logging` for example. Should I always dig into github sources?
> 
>     * If I know exact version numbers, how do I construct links for `wit/deps.toml` ? Is there any rule or hint?

These are all valid points, `wasi:logging` is a draft proposal and up until very recently did not even have a version.
Commits before https://github.com/WebAssembly/wasi-logging/commit/d31c41d0d9eed81aabe02333d0025d42acf3fb75 would work with the latest released host and support for the *versioned* logging, introduced in the aforementioned commit is going to be available in next release (already merged in `main` #3281 )

For best results, I'd suggest sticking with https://github.com/WebAssembly/wasi-logging/commit/3293e84de91a1ead98a1b4362f95ac8af5a16ddd for now.

For all WASI draft interfaces, we support all the latest draft, so you should be able to use the latest tag, e.g. `v0.2.0-draft` for `wasi:blobstore` https://github.com/WebAssembly/wasi-blobstore/tree/v0.2.0-draft.

`wasi:config` recently undergone a rename, so similarly to `wasi:logging` you either need to use `main`/wait until next wasmCloud release or pin to a commit https://github.com/WebAssembly/wasi-config/commit/c667fe67ffa158925511810775155af582d22928

Due to the unstable nature of these draft WASI proposals, we cannot really provide any stability guarantees around them, but we are currently working on a way to mitigate this and "insulate" users from the breaking changes in this regard, expect updates on that very soon.

In terms of constructing the URLs - these are just regular GitHub URLs. GitHub automatically provides source code of a repository to be downloaded via `<repo>/archive/<spec>.tar.gz`, e.g. https://github.com/WebAssembly/wasi-keyvalue/archive/v0.2.0-draft.tar.gz or https://github.com/WebAssembly/wasi-keyvalue/archive/219ea3612a53f1bf5b2d137551b22d0268fd3c58.tar.gz or even a branch name (like `main`). These URLs are really just links to tarballs.

See https://github.com/wasmCloud/wasmCloud/pull/3334 for example dependency specification fixes, perhaps that could be a good reference

By the way, we are currently working on adopting https://github.com/bytecodealliance/wasm-pkg-tools as a replacement for `wit-deps`, which will simplify dependency management here and provide a more streamlined experience

**Comment by deem0n** [comment]
Thanks for the detailed answer! I guess we can close the issue now?
